### PR TITLE
Updated the documentation to split it into smaller chapters and fixed some RST syntax issues

### PR DESCRIPTION
**Pull request by javiereguiluz**

### CODE REVIEW

File: Resources/doc/index.rst [hunk 1]
```diff
@@ -1,334 +1,42 @@
-Provides integration for DoctrineExtensions_ for your Symfony2 Project.
+StofDoctrineExtensions
```
Comment (stof) [inline Resources/doc/index.rst]: the title is wrong. There is no StofDoctrineExtensions, only a StofDoctrineExtensionsBundle

Comment (javiereguiluz) [inline Resources/doc/index.rst]: Sorry! Fixed.

**Comment by stof** [comment]
@javiereguiluz I merged your other PR updated the doc. Can you rebase this one to fix conflicts ?

**Comment by JHGitty** [comment]
ping @javiereguiluz

**Comment by javiereguiluz** [comment]
@JHGitty rebased!

**Comment by javiereguiluz** [comment]
Is this PR still relevant or you prefer to close it?

File: Resources/doc/advanced.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                sluggable:      ~
+                translatable:   ~
+                loggable:       ~
+                softdeleteable: ~
+                uploadable:     ~
+
+    .. code-block:: xml
+
+        <!-- app/config/config.xml -->
+        <container xmlns:doctrine_extensions="http://symfony.com/schema/dic/stof_doctrine_extensions">
```
Comment (stof) [inline Resources/doc/advanced.rst]: the alias defined here is not consistent with the one used in tags just below

File: Resources/doc/installation.rst [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+Step 1: Download the Bundle
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Open a command console, enter your project directory and execute the
+following command to download the latest stable version of this bundle:
+
+.. code-block:: bash
+
+    $ composer require stof/doctrine-extensions-bundle:"~1.1@dev"
```
Comment (stof) [inline Resources/doc/installation.rst]: no consistent with the sentence above talking about *stable* version

File: Resources/doc/index.rst [hunk 4]
```diff
@@ -1,352 +1,42 @@
...
+    installation
+    configuration
+    softdeletable-filter
+    uploadable-extension
+    advanced
 
 .. _`DoctrineExtensions`: https://github.com/Atlantic18/DoctrineExtensions
-.. _`DoctrineExtensions documentation`: https://github.com/Atlantic18/DoctrineExtensions/tree/master/doc/
-.. _`installation chapter`: https://getcomposer.org/doc/00-intro.md
+.. _`official DoctrineExtensions documentation`: https://github.com/Atlantic18/DoctrineExtensions/tree/master/doc/
```
Comment (stof) [inline Resources/doc/index.rst]: should use the 2.4.x branch, as done in master now.

File: Resources/doc/uploadable-extension.rst [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+        $this->redirect($this->generateUrl(...));
+    }
+
+    return $this->render('...', array('form' => $form->createView()));
+
+And that's it. The Uploadable extension handles the rest of the stuff. Remember
+to read its documentation!
+
+.. _`official DoctrineExtensions documentation`: https://github.com/Atlantic18/DoctrineExtensions/tree/master/doc/
```
Comment (stof) [inline Resources/doc/uploadable-extension.rst]: saùe here

File: Resources/doc/uploadable-extension.rst [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        $em->persist($document);
+
+        $uploadableManager = $this->get('stof_doctrine_extensions.uploadable.manager');
+
+        // Here, "getMyFile" returns the "UploadedFile" instance that the form bound in your $myFile property
+        $uploadableManager->markEntityToUpload($document, $document->getMyFile());
+
+        $em->flush();
+
+        $this->redirect($this->generateUrl(...));
```
Comment (stof) [inline Resources/doc/uploadable-extension.rst]: missing `return`

**Comment by stof** [comment]
@javiereguiluz it might be great to resurrect this PR (and potentially to add a note in the installation chapter to say that registering the bundle is done automatically by Flex)

**Comment by javiereguiluz** [comment]
I couldn't recover this old PR because I had deleted my fork ... so I created #363 where I fix all the issues you mentioned here. Thanks!

**Pull request closed without merging**